controllers/bindings/targets.go

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bindings
+
+import (
+	"context"
+
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SecretDeploymentTarget, together with SecretBuilder and ObjectMarker, represents a method of obtaining enough
+// information for the DependentsHandler to be able to deliver the secrets and service accounts to some "target"
+// place in (some) K8s cluster.
+type SecretDeploymentTarget interface {
```

Comment:
Why do we need it as an interface?
My understanding it is considered to have 3 implementations (same-namespace, other-namespace, other-cluster) ?
First 2 differs by namespace name, third - (will) have other client... Or I misunderstand something?

Reply:
The interface hides the different mechanics of obtaining the namespace and the client.

Reply:
I'd avoid doing that in the core model assuming that business logic on top can manage this mapping, we Probably can imagine it as a pluggable mechanism, but let's start simply.